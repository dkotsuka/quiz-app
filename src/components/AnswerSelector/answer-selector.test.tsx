import React from 'react'
import { AnswerSelector } from '.'
import {create, act} from 'react-test-renderer';

describe("AnswerSelector", () => {
    const mockFn = jest.fn()
    const component = create(<AnswerSelector text="My answer" isSelected={true} onPress={mockFn}/>)
    
    it("match snapshot", () => {
        expect(component).toMatchSnapshot()
    })

    it("call the function when pressed", () => {
        const touchable = component.root.findByProps({testID:"answer-selector-touchable"}).props
        act(() => touchable.onPress())
        act(() => touchable.onPress())
        act(() => touchable.onPress())

        expect(mockFn).toBeCalled()
        expect(mockFn).toBeCalledTimes(3)
    })

})