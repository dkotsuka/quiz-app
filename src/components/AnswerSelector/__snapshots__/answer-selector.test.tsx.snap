// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AnswerSelector match snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#FFFFFF",
      "borderColor": "#4D8AF0",
      "borderRadius": 8,
      "borderWidth": 2,
      "elevation": 3,
      "marginBottom": 16,
      "opacity": 1,
      "paddingBottom": 17,
      "paddingLeft": 17,
      "paddingRight": 17,
      "paddingTop": 17,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.4,
      "shadowRadius": 3,
    }
  }
  testID="answer-selector-touchable"
>
  <Text
    size={14}
    style={
      Array [
        Object {
          "fontFamily": "Roboto",
          "fontSize": 14,
        },
      ]
    }
  >
    My answer
  </Text>
</View>
`;
